### ISSUE TITLE
Error Handling: Checkbox Unchecked and Incorrect Message When Unlinking In-Use Shipment Method from Carrier

### ISSUE THREAD

**Issue by Adameddy26** [hotwax/fulfillment#707]
## Current behavior
The checkbox is unchecked when attempting to unlink the shipment method.
The error toast message is incorrect or unclear, failing to communicate the reason for the issue effectively.

## Expected behavior
Checkbox State: The checkbox for the shipment method should remain checked if it is currently used for an order. It should not be unchecked when an attempt is made to unlink it.
Error Toast Message: A clear and sensible error message should be displayed to inform the user that the shipment method cannot be unlinked because it is in use for an order. The message should guide the user on what actions they can take or provide an explanation for the restriction.

## Steps to reproduce the issue
Select a carrier and a shipment method for an order.
Try to unlink or deselect the shipment method from the carrier.
Observe the behavior of the checkbox and the error toast message.

## Environment details
App version : v2.7.2
App environment : UAT

## Additional information
https://jam.dev/c/fa992167-94e2-4b6b-9bbc-eb83c4d12ee1

**Comment by dt2patel** [comment]
@Adameddy26 do you have a suggesting message here?

**Comment by Adameddy26** [comment]
> @Adameddy26 do you have a suggesting message here?

Cannot unlink: Shipment method is in use for active orders.

**Comment by dt2patel** [comment]
@priyanshee1604 If a carrier shipment method gets applied to a shipment once, does that mean we cannot remove that carrier shipment method data ever?

**Comment by priyanshee1604** [comment]
> [@priyanshee1604](https://github.com/priyanshee1604) If a carrier shipment method gets applied to a shipment once, does that mean we cannot remove that carrier shipment method data ever?

Yes, when a carrier shipment method is applied to a shipment  we cannot delete the shipment method data due to the forgein key constraint on the various invovled entities in the shipment data modal.

**Comment by ravilodhi** [comment]
@ashutosh7i Handle the UI part to check/uncheck the checkbox only if the API returns success.

**Comment by ravilodhi** [comment]
UI is fixed to handle checkbox state correctly based on the API response.

We cannot delete a carrier shipment method once it has been applied, primarily due to foreign key constraints in the OrderItemShipGroup table. It also has foreign key references in the FacilityCarrierShipment, ShoppingList, and ShippingCostEstimate tables.

Also if we want to correct the error message, every time we need to check in DB (in OrderItemShipGroup table) if there is any order with that carrier shipment method. 

@dt2patel What should we do here?

**Comment by dt2patel** [comment]
For error message, I don't understand why we need to check DB. If there is a foreign key constraint then system will return error right? It will give error that ID is already in use and so cannot be deleted?

**Comment by ravilodhi** [comment]
The error message returned by the entity-auto services are not user friendly. Here is the API response - 

{
    "error": {
        "code": 500,
        "message": "Error doing entity-auto operation for entity CarrierShipmentMethod in service deleteCarrierShipmentMethod: org.apache.ofbiz.entity.GenericEntityException: org.apache.ofbiz.entity.GenericDataSourceException: Exception while deleting the following entity: [GenericEntity:CarrierShipmentMethod][partyId,FEDEX(java.lang.String)][roleTypeId,CARRIER(java.lang.String)][shipmentMethodTypeId,NEXT_DAY(java.lang.String)] (SQL Exception while executing the following:DELETE FROM CARRIER_SHIPMENT_METHOD WHERE SHIPMENT_METHOD_TYPE_ID=? AND PARTY_ID=? AND ROLE_TYPE_ID=? (Cannot delete or update a parent row: a foreign key constraint fails (`devhc`.`order_item_ship_group`, CONSTRAINT `ORDER_ITSG_CSHM` FOREIGN KEY (`SHIPMENT_METHOD_TYPE_ID`, `CARRIER_PARTY_ID`, `CARRIER_ROLE_TYPE_ID`) REFERENCES `carrier_shipment_method` (`SHIPMEN))) (Exception while deleting the following entity: [GenericEntity:CarrierShipmentMethod][partyId,FEDEX(java.lang.String)][roleTypeId,CARRIER(java.lang.String)][shipmentMethodTypeId,NEXT_DAY(java.lang.String)] (SQL Exception while executing the following:DELETE FROM CARRIER_SHIPMENT_METHOD WHERE SHIPMENT_METHOD_TYPE_ID=? AND PARTY_ID=? AND ROLE_TYPE_ID=? (Cannot delete or update a parent row: a foreign key constraint fails (`devhc`.`order_item_ship_group`, CONSTRAINT `ORDER_ITSG_CSHM` FOREIGN KEY (`SHIPMENT_METHOD_TYPE_ID`, `CARRIER_PARTY_ID`, `CARRIER_ROLE_TYPE_ID`) REFERENCES `carrier_shipment_method` (`SHIPMEN))))"
    }
}

**Comment by shubham-namdeo** [comment]
Checkbox UI is functioning as per the requirement.

**Comment by Varnita-Jain** [comment]
<img width="1851" height="944" alt="Image" src="https://github.com/user-attachments/assets/d7051afd-ec7f-4e22-b5c2-93c03a86b4c2" /> - working as expected 


Verified – the shipment method checkbox remains checked when it is in use for an order, and a clear error message is displayed if an attempt is made to unlink it. Issue fixed and working as expected.
